src/RenderWebGL.js

```diff
@@ -737,6 +737,40 @@ class RenderWebGL extends EventEmitter {
     }
 
     /**
+     * Update the position object's x & y members to keep the drawable fenced in view.
+     * @param {int} drawableID - The ID of the Drawable to update.
+     * @param {object} position - An object with x & y members to be fenced
+     *                          - These values are updated in the object to become fenced
+     */
+    getFencedPositionOfDrawable (drawableID, position) {
+        const drawable = this._allDrawables[drawableID];
+        if (!drawable) {
+            // TODO: fix whatever's wrong in the VM which causes this, then add a warning or throw here.
+            // Right now this happens so much on some projects that a warning or exception here can hang the browser.
+            return;
+        }
+        const dx = position.x - drawable._position[0];
+        const dy = position.y - drawable._position[1];
+
+        const aabb = drawable.getFastBounds();
+
+        // This is my best guess at the fencing in Scratch 2,
+        // but I suspect it may need further work to be precisely the same?
```

Comment:
The Scratch 2.0 implementation can be found in the function `keepOnStage` in ScratchSprite.as:
https://github.com/LLK/scratch-flash/blob/6651badebc4b4ee9d63ad7565bae27c5cb6611d0/src/scratch/ScratchSprite.as#L191

It looks to me like your implementation should have the same effect for a sprite with a non-empty costumes. In particular, 18 - 3 = 15.

If I'm reading the Scratch 2.0 code right, for a sprite using an empty costume -- where `width == height == 0` -- it effectively moves the fence outward by 3 units. I can't think of a reason for that... can you? If not, I'm happy with dropping that inconsistency until we find a reason to bring it back.